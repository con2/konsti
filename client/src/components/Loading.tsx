import React, { FC, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import loaderImage from 'assets/loading.gif';

export const Loading: FC = (): ReactElement => {
  const { t } = useTranslation();
  return (
    <LoadingContainer>
      <p>{t('loading')}</p>
      <img alt={t('loading')} src={loaderImage} width='40' />
    </LoadingContainer>
  );
};

const LoadingContainer = styled.div`
  display: flex;
  flex: 1 0 auto;
  flex-direction: column;
  align-items: center;
  font-weight: 600;
`;
