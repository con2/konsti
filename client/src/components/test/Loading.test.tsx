import React from 'react';
import { shallow } from 'enzyme';
import { Loading } from '../Loading';

describe('Loading', () => {
  it('should render correctly', () => {
    const component = shallow(<Loading />);
    expect(component).toMatchSnapshot();
  });
});
