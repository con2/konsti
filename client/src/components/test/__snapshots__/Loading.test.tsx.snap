// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<Loading__LoadingContainer>
  <p>
    loading
  </p>
  <img
    alt="loading"
    src=""
    width="40"
  />
</Loading__LoadingContainer>
`;
